crates/api/src/v1/fetch.rs

```diff
@@ -53,8 +64,8 @@ pub enum FetchError {
     #[error("log `{0}` was not found")]
     LogNotFound(LogId),
     /// The provided record was not found.
-    #[error("record `{0}` was not found")]
-    RecordNotFound(RecordId),
+    #[error("record fetch token `{0}` was not found")]
+    RecordNotFound(String),
```

Comment:
Maybe `InvalidFetchToken`? Also do we even need this? Now that the client can handle it we might just want to return the entire log on invalid fetch token to allow changes to the token format.

Reply:
I think an error makes sense. Happy to rename it to `InvalidFetchToken`.

Reply:
How would a server migrate fetch token formats?

Reply:
With this PR, as is:

TLDR: it would work just fine.

The server implementation isn't required to return an error for `InvalidFetchToken`. If it does not error and returns more records than necessary, the client will ignore the records that it already has (without updating to the new `fetchToken` for the previous records). Once a new record is encountered, that new `fetchToken` is now saved and used for future pagination.

I would assume that a server implementation that is migrating to a new `fetchToken` format would not error and would still support the old format (detecting the difference if format versions) for some period of time.

Reply:
I'll see if I can push a change that makes the client more resilient for `fetchToken` migration.

Reply:
We can discuss on the Registry call, but this implementation is likely sufficiently resilient as is.

Reply:
Renamed the `RecordNotFound` for the `FetchError` to `FetchTokenNotFound`. Hope that feels better.